src/backend/commands/createas.c

```diff
@@ -1075,9 +1095,15 @@ check_ivm_restriction_walker(Node *node, check_ivm_restriction_context *ctx, int
 				foreach(lc, qry->rtable)
 				{
 					RangeTblEntry *rte = (RangeTblEntry *) lfirst(lc);
+
+					if (rte->tablesample != NULL)
+						ereport(ERROR, (errmsg("TABLESAMPLE parameter is not supported with IVM")));
+					if (rte->relkind == RELKIND_RELATION && has_subclass(rte->relid))
```

Comment:
has_subcalss() can return a false-positive result, so I think we should use find_inheritance_children(). We will not care the overhead to check pg_inherits during creating IMMV.  We would not want it if we did it for every query execution.

Reply:
Thank you for your comments.
My understanding is that find_inheritance_children() must be used with 'AccessShareLock' lockmode. Is this correct?

Reply:
Could you please explain why you think so?

Reply:
My understanding is that
this function return  false-positive result too, if DROP command is executed for child table by other session and Nolock is specified.  Because inheritance children is not locked,  same as has_sublclass() result.

When we use other than NoLock, this function wait for finished DROP, and detect no relation.

Reply:
Could it be a problem in the real situation? 

> this function return false-positive result too, if DROP command is executed for child table by other session and Nolock is specified

In this case, we will conclude that it is inheritance parent at this time, and creating IMMV will fail. However, I think this would not  be a problem because this will work after the dropping session is finished. However, if we use has_subclass, we will need ANALYZE (or autoanalyze) before creating IMMV, even after their child tables are all dropped.  I think this is confusable for users, so it seems better to for me use find_inheritance_children for me as longer as this is not difficult for some reasons.

Reply:
Thank you for your explanation. I misinterpreted behavior of has_sublclass() and  find_inheritance_children().
Ok, I will fix to use  find_inheritance_children().